### PR TITLE
[WIP] Release/5.0.0-beta.1

### PR DESCRIPTION
**Pull request by joshdholtz**
The first beta of **RevenueCat Purchases Hybrid Common v5** is here!! 😻

This latest release updates the Android SDK dependency from v5 to [v6](https://github.com/RevenueCat/purchases-android/releases/tag/6.0.0) to use BillingClient 5. This version of BillingClient brings an entire new subscription model which has resulted in large changes across the entire SDK.

### Migration Guides
- See [Android Native - 5.x to 6.x Migration](https://www.revenuecat.com/docs/android-native-5x-to-6x-migration) for a
  more thorough explanation of the new Google subscription model announced with BillingClient 5 and how to take
  advantage of it in V6. This guide includes tips on product setup with the new model.

#### Purchasing
In v4, a Google Play Android `Package` or `StoreProduct` represented a single purchaseable entity, and free trials or intro
offers would automatically be applied to the purchase if the user was eligible.

Now, in Hybrid Common v5, an Google Play Android `Package` or `StoreProduct` represents a duration of a subscription and contains all the ways to
purchase that duration -- any offers and its base plan. Each of these purchase options are `SubscriptionOption`s.
When passing a `Package` to `purchasePackage()` or `StoreProduct` to `purchaseStoreProduct()`, the SDK will use the following logic to choose which
`SubscriptionOption` to purchase:
- Filters out offers with "rc-ignore-offer" tag
- Uses `SubscriptionOption` with the longest free trial or cheapest first phase
    - Only offers the user is eligible will be applied
- Falls back to base plan

For more control, purchase subscription options with the new `purchaseSubscriptionOption()` method.

#### Models

`StoreProduct` now has a few new properties use for Google Play Android:
- `defaultOption`
  - A subscription option that will automatically be applie when purchasing a `Package` or `StoreProduct`
- `subscriptionOptions`
  - A list of subscription options (could be null)

##### Subscription Option

Below is an example of what a subscription option:

```js
{
    "id": "basePlan",
    "storeProductId": "subId:basePlanId",
    "productId": "subId",
    "pricingPhases": [
        {
            "price": 0,
            "priceString": "FREE",
            "period": "P1M",
            "cycles": 1
        },
        {
            "price": 4.99,
            "priceString": "$4.99",
            "period": "P1M",
            "cycles": 0
        }
    ],
    "tags": ["free-offers"],
    "isBasePlan": false,
    "billingPeriod": {
        "periodUnit": "MONTH",
        "periodNumberOfUnits": 0
    },
    "fullPricePhase": {
        "price": 4.99,
        "priceString": "$4.99",
        "period": "P1M",
        "cycles": 0
    },
    "freePhase" {
        "price": 0,
        "priceString": "FREE",
        "period": "P1M",
        "cycles": 1
    },
    "introPhase": null
}
```

### Observer Mode

Observer mode is still supported in v5. Other than updating the SDK version, there are no changes required.

### CODE REVIEW

**Review by tonidero — APPROVED** [review]
Just improve that changelog, but otherwise it's good!

File: CHANGELOG.latest.md [hunk 1]
```diff
@@ -1,2 +1,79 @@
...
+### Migration Guides
+- See [Android Native - 5.x to 6.x Migration](https://www.revenuecat.com/docs/android-native-5x-to-6x-migration) for a
+  more thorough explanation of the new Google subscription model announced with BillingClient 5 and how to take
+  advantage of it in V6. This guide includes tips on product setup with the new model.
+
+### New `SubscriptionOption` concept
+
+#### Purchasing
+
+Now, in Hybrid Common v5, an Google Play Android `Package` or `StoreProduct` represents a duration of a subscription and contains all the ways to
```
Comment (beylmk) [inline CHANGELOG.latest.md:14]: i think we had a "before" line describing how the offers used to get automatically attached?

File: CHANGELOG.latest.md [hunk 2]
```diff
@@ -1,2 +1,79 @@
...
+purchase that duration -- any offers and its base plan. Each of these purchase options are `SubscriptionOption`s.
+When passing a `Package` to `purchasePackage()` or `StoreProduct` to `purchaseStoreProduct()`, the SDK will use the following logic to choose which
+`SubscriptionOption` to purchase:
+*   - Filters out offers with "rc-ignore-offer" tag
+*   - Uses `SubscriptionOption` with the longest free trial or cheapest first phase
+*   - Falls back to base plan
```
Comment (beylmk) [inline CHANGELOG.latest.md:20]: one thing i was thinking about (for native too) is that maybe we should explain we'll never apply an offer the user isn't eligible for...

our customers shouldn't have to know what google passes to us, and i don't think it's immediately clear

File: CHANGELOG.latest.md [hunk 3]
```diff
@@ -1,2 +1,79 @@
...
+
+For more control, purchase subscription options with the new `purchaseSubscriptionOption()` method.
+
+#### Models
+
+`StoreProducts` now has a few new properties use for Google Play Android:
```
Comment (beylmk) [inline CHANGELOG.latest.md:26]: nit: `StoreProduct`

File: CHANGELOG.latest.md [hunk 4]
```diff
@@ -1,2 +1,79 @@
...
+- `defaultOption`
+  - A subscription option (could be null)
```
Comment (beylmk) [inline CHANGELOG.latest.md:28]: maybe "that we will automatically apply when purchasing a `Product` or `Package`"

File: CHANGELOG.latest.md (hunk 1 above)
Comment (joshdholtz) [reply CHANGELOG.latest.md:14]: Oh, I missed that 🤦‍♂️

**Review by beylmk — APPROVED** [review]
🚢

File: ios/PurchasesHybridCommon/Info.plist [hunk 5]
```diff
@@ -15,7 +15,7 @@
 	<key>CFBundlePackageType</key>
 	<string>$(PRODUCT_BUNDLE_PACKAGE_TYPE)</string>
 	<key>CFBundleShortVersionString</key>
-	<string>4.14.1</string>
```
Comment (beylmk) [inline ios/PurchasesHybridCommon/Info.plist:18]: does this one need to have the beta in it?

**Pull request merged**